File: stocks-api/src/shared/env.go
package shared

import (
	"fmt"
	"log"
	"os"
	"strconv"

	"github.com/joho/godotenv"
)

// Configurations that need the server.
type config struct {
	DbConnection string
	Database string
	UrlApi string
	Port int
	SecurityApi string
	Secure bool
	MqServerUrl string
} 

// Get variable configurations with its respective type that has been setted via environment variables or from .env file.
// Those variables is use along of the system.
func GetConfig() *config{
	if os.Getenv("GO_ENV") == "develop" || os.Getenv("GO_ENV") == "" {
		loadDotEnv()
	}
	return &config{
		DbConnection: os.Getenv("DB_CONNECTION"),
		Database: os.Getenv("DATABASE"),
		UrlApi: getUrlApi(),
		Port: getPort(),
		SecurityApi: os.Getenv("SECURITY_API_URL"),
		Secure: os.Getenv("SECURE") ==  "true",
		MqServerUrl: getMqServerUrl(),
	}
}

// Get variables from .env files.
func loadDotEnv(){
	err := godotenv.Load()
	if err != nil {
		log.Fatalln("Error loading .env file")
	}
}

// Get URL API string either from env variable or a default URL.
func getUrlApi() string{
	urlApi, ok := os.LookupEnv("URL_API")
	if !ok {
		return fmt.Sprintf("http://localhost:%d",getPort())
	}
	return urlApi
}

// Get PORT API number either from env variable or a default PORT.
func getPort() int{
	portStr, ok := os.LookupEnv("PORT")
	if !ok {
		return 8081
	}
	port , _ := strconv.Atoi(portStr)
	return port
}

func getMqServerUrl() string {
	urlMq, ok := os.LookupEnv("MQ_SERVER_URL")
	if !ok {
		return "tcp://127.0.0.1:9000"
	}
	return urlMq
}

File: payment-api/src/shared/env/env.ts
import dotenv from 'dotenv';

process.env.NODE_ENV = process.env.NODE_ENV || 'development';

dotenv.config();

export default {
  NODE_ENV: process.env.NODE_ENV,
  PORT: Number(process.env.PORT || 8080),
  DOCS_ENDPOINT: '/docs',
  DIR_SWAGGER: process.env.DIR_SWAGGER || './src/shared/docs/swagger.yml',
  DB_CONNECTION: process.env.DB_CONNECTION || 'localhost',
  SECURITY_API_URL: process.env.SECURITY_API_URL || 'http://localhost:8080',
  PREFIX_URL: process.env.PREFIX_URL || '/',
  MQ_SERVER_URL: process.env.MQ_SERVER_URL || 'tcp://127.0.0.1:9000',
};
